### PR TITLE
Improve ObjC bazel test and add TODOs.

### PR DESCRIPTION
**Pull request by jtattermusch**
- run more tests as part of the objc bazel test job
- clarify the situation with setting GRPC_VERBOSITY when using bazel test (turns out GRPC_VERBOSITY=debug is already being applied to the tests).
- add many TODOs to the run_tests.py section that defines ObjC test tasks. These TODOs pretty much determine what should be the next steps in ObjC test cleanup.

The ObjC bazel test job is now running both continuously on master, as well as presubmit on PRs, so we are now in good shape for others to step in and address all the TODOs. Just to make sure we're on the same page, `TODO(jtattermusch)` convention means that jtattermusch has created the TODO item, not that I'm going to be the one who addresses it.

### CODE REVIEW

**Comment by jtattermusch** [comment]
CC @HannahShiSFB

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -1038,70 +1042,88 @@ def test_specs(self):
...
+        # it doesn't really build any ObjC code.
+        # TODO(jtattermusch): turn this test into a bazel test or come up with a better place where to put this test.
         out.append(
             self.config.job_spec(
                 ['src/objective-c/tests/run_plugin_option_tests.sh'],
                 timeout_seconds=60 * 60,
                 shortname='ios-test-plugin-option-test',
                 cpu_cost=1e6,
                 environ=_FORCE_ENVIRON_FOR_WRAPPERS))
+        # TODO(jtattermusch): move the test out of the test/core/iomgr/CFStreamTests directory?
```
[the remark this answers is not in the document]
Comment (dennycd) [reply tools/run_tests/run_tests.py:1063]: these are test suite for iOS's CFStream iomgr fork which are implemented as part of the c-core. Ideally we should move these out of test/core and have them co-located with other iOS test suites.

File: tools/run_tests/run_tests.py [hunk 2]
```diff
@@ -1038,70 +1042,88 @@ def test_specs(self):
...
+        # TODO(jtattermusch): Clarify what's the difference between PerfTests and PerfTestsPosix
+        # TODO(jtattermusch): Create bazel target for the test and remove the test from here.
         out.append(
             self.config.job_spec(['src/objective-c/tests/run_one_test.sh'],
                                  timeout_seconds=30 * 60,
                                  shortname='ios-perf-test-posix',
                                  cpu_cost=1e6,
                                  environ={'SCHEME': 'PerfTestsPosix'}))
+        # TODO(jtattermusch): Create bazel target for the test (how does one add the cronet dependency in bazel?)
+        # TODO(jtattermusch): move the test out of the test/cpp/ios directory?
```
[the remark this answers is not in the document]
Comment (dennycd) [reply tools/run_tests/run_tests.py:1119]: similar to CFStream test suite, these are for Cronet transport test run under iOS test environment. Will add a task to follow up and move these out of test/cpp/ios

**Review by dennycd — APPROVED** [review]
thanks Jan, I will create follow up issues to address these todos. Also free free to directly add me to it : )

**Pull request merged**